File: [6 kyu] Multiples of 3 or 5/main.py
def solution(number):
    max3 = (number - 1) // 3
    max5 = (number - 1) // 5
    max15 = (number - 1) // 15

    sum3 = 0 if max3 <= 0 else (1 + max3) * max3 * 3 // 2
    sum5 = 0 if max5 <= 0 else (1 + max5) * max5 * 5 // 2
    sum15 = 0 if max15 <= 0 else (1 + max15) * max15 * 15 // 2

    return sum3 + sum5 - sum15


def solution(number):
    return sum(num for num in range(number) if num % 3 == 0 or num % 5 == 0)


def solution(number):
    return sum(set(range(3, number, 3)) | set(range(5, number, 5)))

File: [6 kyu] Multiples of 3 or 5/main.cpp
int solution(int number) {
    long long max3 = (number - 1) / 3;
    long long max5 = (number - 1) / 5;
    long long max15 = (number - 1) / 15;

    long long sum3 = (max3 <= 0) ? 0 : ((1 + max3) * max3 * 3 / 2);
    long long sum5 = (max5 <= 0) ? 0 : ((1 + max5) * max5 * 5 / 2);
    long long sum15 = (max15 <= 0) ? 0 : ((1 + max15) * max15 * 15 / 2);

    return (int)(sum3 + sum5 - sum15);
}

int solution(int number) {
    int sum = 0;

    for (int i = 1; i < number; i++) {
        if (i % 3 == 0 || i % 5 == 0) {
            sum += i;
        }
    }

    return sum;
}